redfish-core/lib/account_service.hpp

```diff
@@ -1288,177 +1288,187 @@ inline void handleLDAPPatch(nlohmann::json& input,
+        }
+        else if (retval != PAM_SUCCESS)
+        {
+            messages::internalError(asyncResp->res);
             return;
         }
-
-        if (password)
+        else
         {
-            int retval = pamUpdatePassword(username, *password);
-
-            if (retval == PAM_USER_UNKNOWN)
-            {
-                messages::resourceNotFound(asyncResp->res, "ManagerAccount",
-                                           username);
-            }
-            else if (retval == PAM_AUTHTOK_ERR)
-            {
-                // If password is invalid
-                messages::propertyValueFormatError(asyncResp->res, nullptr,
-                                                   "Password");
-                BMCWEB_LOG_ERROR("pamUpdatePassword Failed");
-            }
-            else if (retval != PAM_SUCCESS)
-            {
-                messages::internalError(asyncResp->res);
-                return;
-            }
-            else
-            {
-                // Remove existing sessions of the user when password changed
-                persistent_data::SessionStore::getInstance()
-                    .removeSessionsByUsernameExceptSession(username, session);
-                messages::success(asyncResp->res);
-            }
+            // Remove existing sessions of the user when password changed
+            persistent_data::SessionStore::getInstance()
+                .removeSessionsByUsernameExceptSession(username, session);
+            messages::success(asyncResp->res);
         }
+    }
 
-        if (mfaBypass)
+    if (mfaBypass)
+    {
+        std::string mfaBypassDbusVal;
+        // Check if the input vector contains just one bypass type:
+        // as there are just two defined in the backend:
+        // GoogleAuthenticator  and None
+        if (username == "service")
+        {
+            messages::operationNotAllowed(asyncResp->res);
+            return;
+        }
+        if (mfaBypass->size() > 1)
+        {
+            std::string values = std::accumulate(
+                std::next(mfaBypass->begin()), mfaBypass->end(),
+                mfaBypass->front(),
+                [](const std::string& str1, const std::string& str2) {
+                return str1 + ", " + str2;
+            });
+            messages::propertyValueIncorrect(asyncResp->res,
+                                             "MFABypass/BypassTypes", values);
+            return;
+        }
+        std::string mfaBypassDbusPrefix =
+            "xyz.openbmc_project.User.MultiFactorAuthConfiguration.Type.";
+        if (mfaBypass->empty())
         {
-            std::string mfaBypassDbusVal;
-            // Check if the input vector contains just one bypass type:
-            // as there are just two defined in the backend:
-            // GoogleAuthenticator  and None
-            if (username == "service")
+            mfaBypassDbusVal = mfaBypassDbusPrefix + "None";
```

Comment:
Would this be better to use the latest upstream patchset ?
https://gerrit.openbmc.org/c/openbmc/bmcweb/+/74667/37/redfish-core/lib/account_service.hpp#1162
```
            mfaBypassDbusVal =
                "xyz.openbmc_project.User.MultiFactorAuthConfiguration.Type.None";
```

Reply:
yes. will change it in my next PS.

Reply:
Done